scm-webapp/src/main/java/sonia/scm/api/v2/resources/UserApiKeyResource.java

```diff
@@ -0,0 +1,192 @@
+    content = @Content(
+      mediaType = VndMediaType.ERROR_TYPE,
+      schema = @Schema(implementation = ErrorDto.class)
+    ))
+  @ApiResponse(
+    responseCode = "500",
+    description = "internal server error",
+    content = @Content(
+      mediaType = VndMediaType.ERROR_TYPE,
+      schema = @Schema(implementation = ErrorDto.class)
+    ))
+  public ApiKeyDto get(@PathParam("id") String id, @PathParam("keyId") String keyId) {
+    return apiKeyService
+      .getKeys(id)
+      .stream()
+      .filter(key -> key.getId().equals(keyId))
+      .map(key -> apiKeyMapper.map(key, id))
+      .findAny()
+      .orElseThrow(() -> notFound(ContextEntry.ContextBuilder.entity(ApiKey.class, keyId)));
+  }
+
+  @POST
+  @Path("")
+  @Consumes(VndMediaType.API_KEY)
+  @Produces(MediaType.TEXT_PLAIN)
+  @Operation(
+    summary = "Create new api key for user",
+    description = "Creates a new api key for the given user with the role specified in the given key.",
+    tags = "User",
+    operationId = "create_api_key",
+    requestBody = @RequestBody(
+      content = @Content(
+        mediaType = VndMediaType.API_KEY,
+        schema = @Schema(implementation = CreateApiKeyDto.class),
+        examples = @ExampleObject(
+          name = "Create a new api key named readKey with READ permission role.",
+          value = "{\n  \"displayName\":\"readKey\",\n  \"permissionRole\":\"READ\"\n}",
+          summary = "Create new api key"
+        )
+      )
+    )
+  )
+  @ApiResponse(
+    responseCode = "201",
+    description = "create success",
+    headers = @Header(
+      name = "Location",
+      description = "uri to the created user",
+      schema = @Schema(type = "string")
+    ),
+    content = @Content(
+      mediaType = MediaType.TEXT_PLAIN
+    )
+  )
+  @ApiResponse(responseCode = "401", description = "not authenticated / invalid credentials")
+  @ApiResponse(responseCode = "409", description = "conflict, a key with the given display name already exists")
+  @ApiResponse(
+    responseCode = "500",
+    description = "internal server error",
+    content = @Content(
+      mediaType = VndMediaType.ERROR_TYPE,
+      schema = @Schema(implementation = ErrorDto.class)
+    ))
+  public Response create(@Valid ApiKeyDto apiKey, @PathParam("id") String id) {
+    final ApiKeyService.CreationResult newKey = apiKeyService.createNewKey(id, apiKey.getDisplayName(), apiKey.getPermissionRole());
+    return Response.status(CREATED)
+      .entity(newKey.getToken())
+      .location(URI.create(resourceLinks.apiKey().self(newKey.getId(), id)))
+      .build();
+  }
+
+  @DELETE
+  @Path("{keyId}")
+  @Operation(summary = "Delete api key", description = "Deletes the api key with the given id for user.", tags = "User", operationId = "delete_api_key")
+  @ApiResponse(responseCode = "204", description = "delete success or nothing to delete")
+  @ApiResponse(responseCode = "401", description = "not authenticated / invalid credentials")
```

Comment:
Same goes for this, unit test please

Reply:
Same goes for this. Checking this is not possible with a unit test since the permission check is encapsuled inside the ApiKeyService.